lib/sinon/util/core/iterable-to-string.js

```diff
@@ -0,0 +1,23 @@
+"use strict";
+var typeOf = require("../../typeOf");
+
+module.exports = function (obj) {
+    var representation = "";
+
+    function stringify(item) {
+        return typeof item === "string" ? "'" + item + "'" : String(item);
+    }
+
+    if (typeOf(obj) === "map") {
```

Comment:
In my eyes `iterableToString` is a higher abstraction function that covers two input scenarios, `Map` or generic input.

I would suggest de-structuring the `iterableToString` function to call lower abstraction functions, perhaps something like this.

```
"use strict";

var typeOf = require("../../typeOf");

function stringify(item) {
    return typeof item === "string" ? "'" + item + "'" : String(item);
}

function mapToString(map) {
    var representation = "";

    map.forEach(function (value, key) {
        representation += "[" + stringify(key) + "," + stringify(value) + "],";
    });

    representation = representation.slice(0, -1);

    return representation;
}

function genericIterableToString(obj) {
    var representation = "";

    obj.forEach(function (value) {
        representation += stringify(value) + ",";
    });

    representation = representation.slice(0, -1);

    return representation;
}


module.exports = function iterableToString(obj) {
    if (typeOf(obj) === "map") {
        return mapToString(obj);
    }

    return genericIterableToString(obj);
};
```